File: class16/python/AP1610-ValidaTriangulo.py
# ValidaTriangulo — Aula 16
# Verifica se três lados formam um triângulo (AND com 3 condições)

a = float(input("Lado A: "))
b = float(input("Lado B: "))
c = float(input("Lado C: "))
if a + b > c and a + c > b and b + c > a:
    print("É um triângulo!")
    if a == b == c:
        print("Tipo: Equilátero")
    elif a == b or a == c or b == c:
        print("Tipo: Isósceles")
    else:
        print("Tipo: Escaleno")
else:
    print("Não forma um triângulo.")

File: class16/cpp/AP1610-ValidaTriangulo.cpp
#include <iostream>
int main() {
    double a, b, c;
    std::cout << "Lado A: "; std::cin >> a;
    std::cout << "Lado B: "; std::cin >> b;
    std::cout << "Lado C: "; std::cin >> c;
    if (a+b>c && a+c>b && b+c>a) {
        std::cout << "É um triângulo!" << std::endl;
        if (a==b && b==c) std::cout << "Tipo: Equilátero" << std::endl;
        else if (a==b || a==c || b==c) std::cout << "Tipo: Isósceles" << std::endl;
        else std::cout << "Tipo: Escaleno" << std::endl;
    } else std::cout << "Não forma um triângulo." << std::endl;
    return 0;
}